Sources/Commands/SwiftTestTool.swift

```diff
@@ -280,20 +294,66 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 numJobs: options.numberOfWorkers ?? ProcessInfo.processInfo.activeProcessorCount,
                 diagnostics: diagnostics,
                 options: self.options,
-                buildParameters: try self.buildParameters()
+                buildParameters: buildParameters
             )
             try runner.run(tests)
 
             if !runner.ranSuccessfully {
                 executionStatus = .failure
             }
+
+            if options.shouldEnableCodeCoverage {
+                try processCodeCoverage()
+            }
+        }
+    }
+
+    /// Function that processes all files in code coverage directory by merging all .profraw files into a .profdata file to export as .json.
+    private func processCodeCoverage() throws {
+        try mergeCodeCovRawDataFiles()
+        let graph = try loadPackageGraph()
+
+        for packages in graph.rootPackages {
+            let buildPlan = try BuildPlan(buildParameters: self.buildParameters(), graph: graph, diagnostics: diagnostics)
+            if options.shouldBuildTests {
+                try build(plan: buildPlan, subset: .allIncludingTests)
```

Comment:
Is this the right place to do this? Presumably, we must have already built and ran the tests to get the raw code-coverage files we merged earlier in this method or am I missing something?